autoload/clap/cache.vim

```diff
@@ -0,0 +1,44 @@
+" Author: Mark Wu <markplace@gmail.com>
+" Description: Cache API for clap.
+
+let s:save_cpo = &cpoptions
+set cpoptions&vim
+
+let s:path_separator = has('win32') ? '\' : '/'
+let s:clap_cache_directory = get(g:, 'clap_cache_directory', '')
```

Comment:
Please also update the doc for `g:clap_cache_directory`.

Reply:
What's the order of help files, It seems not follow alphabet order?

Reply:
Yes, I just insert the new items randomly, putting them in the right section is enough, but feel free to make it more organized, that could be another PR if you are interested.

Reply:
Done!